include/storage.h

```diff
@@ -28,20 +29,22 @@ class storage_t {
         return create_instance(instance_name, script_path);
     }
 
-    seastar::future<bool> run_instance(std::string instance_name) {
+    seastar::future<bool> run_instance(std::string instance_name, std::span<char> data) {
         auto engine_it = v8_instances.find(instance_name);
         if (engine_it == v8_instances.end()) {
             std::cout << "Can not find script " << instance_name << std::endl;
             return seastar::make_ready_future<bool>(false);
         }
 
-        auto run_future = thread_pool.submit([engine_it](){ engine_it->second.run_instance(); });
-        auto stop_future = thread_pool.submit([engine_it](){ engine_it->second.stop_execution_loop(std::chrono::high_resolution_clock::now(), 3.0); });
-
-        return seastar::when_all(std::move(run_future), std::move(stop_future))
-        .then([engine_it](auto res){
-            engine_it->second.continue_execution();
-            return seastar::make_ready_future<bool>(true);
+        return thread_pool.lock()
+        .then([this, engine_it, data]{
+            return engine_it->second.run_instance(thread_pool, 1.0, data);
+        })
+        .then([this](auto res){
+            return thread_pool.unlock()
```

Comment:
It'll be better to do this in the `finally` continuation. If the `run_instance` will return failed future `unlock` won't be called.
Another option is to co_await/co_return in this function and use RAII for semahpore.

Reply:
I think I need to do `lock`/`unlock` inside `run_instance`

Reply:
Done